### ISSUE TITLE
npm install -g azure-cli fails with npm@5

### ISSUE THREAD

**Issue by chrmoritz** [Azure/azure-xplat-cli#3605]
CLI Version: **0.10.13**
OS Type: Mac
Installation via:  both **npm** and **brew** are affected

Mode: (installation fails)

Environment: (installation fails)

Description:
`npm install -g azure-cli` fails when using npm@5 (installed either with `nvm use 8` or the wip  https://github.com/Homebrew/homebrew-core/pull/14085) if not run as root with:
```
npm ERR! path /Users/chris/Desktop/test/azure-xplat-cli-0.10.12-May2017/node_modules/.staging/easy-table-5fa4728c/lib/table.js
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall open
npm ERR! Error: EACCES: permission denied, open '/Users/chris/Desktop/test/azure-xplat-cli-0.10.12-May2017/node_modules/.staging/easy-table-5fa4728c/lib/table.js'
npm ERR!  { Error: EACCES: permission denied, open '/Users/chris/Desktop/test/azure-xplat-cli-0.10.12-May2017/node_modules/.staging/easy-table-5fa4728c/lib/table.js'
npm ERR!   cause: 
npm ERR!    { Error: EACCES: permission denied, open '/Users/chris/Desktop/test/azure-xplat-cli-0.10.12-May2017/node_modules/.staging/easy-table-5fa4728c/lib/table.js'
npm ERR!      errno: -13,
npm ERR!      code: 'EACCES',
npm ERR!      syscall: 'open',
npm ERR!      path: '/Users/chris/Desktop/test/azure-xplat-cli-0.10.12-May2017/node_modules/.staging/easy-table-5fa4728c/lib/table.js' },
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'open',
npm ERR!   path: '/Users/chris/Desktop/test/azure-xplat-cli-0.10.12-May2017/node_modules/.staging/easy-table-5fa4728c/lib/table.js',
npm ERR!   parent: 'azure-cli' }
npm ERR! 
npm ERR! Please try running this command again as root/Administrator.
```

Steps to reproduce:
1) `nvm use 8` or build https://github.com/Homebrew/homebrew-core/pull/14085 from source
2) verify that `npm -v` prints v5.x.x
3) `npm install -g azure-cli` or just `npm install easy-table@0.0.1`

Further informations:

complete npm debug log: https://gist.github.com/chrmoritz/a01420db1b21f347011f0f0b96d0d7f8
Refs: https://github.com/Homebrew/homebrew-core/pull/14085#issuecomment-305385182
and https://github.com/Homebrew/homebrew-core/pull/14085#issuecomment-305592381

npm@<=4 doesn't handle file system permissions inside .tgz at all (it normalized them all to a default one), but npm@5 started to handle/respect them correctly (on unix). Unfortunately the archive of `easy-table@0.0.1` (on which you depend) published to the npm registry has screwed up file system permissions (no +x permission to the lib folder) causing the above permission error when trying to install azure-cli as a non root user with npm@5. This would break the homebrew formula after the npm@5 PR will be merged. I've submitted https://github.com/Homebrew/homebrew-core/pull/14152 as a temporary workaround, which patches your package.json to install `easy-table@0.0.1` from a github tarball instead of the npm registry. But an ideal solution would be if you could update your easy-table dependency to a version with correct file system permission inside the tarball published on npm (looks like the issue was fixed with v0.0.3).

**Comment by balajikris** [comment]
thanks for investigating and suggesting a fix @chrmoritz @DomT4. We'll update our easy-table dependency to a more recent version. Thanks again for reporting the issue and doing all the heavy lifting required to fix the bug 🥇

**Comment by balajikris** [comment]
Fixed with #3608

**Comment by DomT4** [comment]
All credit on identifying the fix here to @chrmoritz. I think I was busy rebuilding `node` for about the 12th time that day to update things at Homebrew by the time Christian had already looked at this 😄. Thanks for fixing it so swiftly though! ❤️
